### PR TITLE
cmd mountlib: adding remount option for linux systems - fixes #6488

### LINKED ISSUE #6488: Add option `--remount` for remount support
#### What is your current rclone version (output from `rclone version`)?
```console
$ rclone --version
rclone 1.59.2
- os/version: nixos 22.11 (Raccoon) (64 bit)
- os/kernel: 5.19.12-zen1 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.18.6
- go/linking: dynamic
- go/tags: cmount
```

#### What problem are you are trying to solve?
I'm using rclone as a mount helper. Whenever I try `sudo systemctl reload <rclone-mount>.mount`, it fails because rclone doesn't recognize the flag `--remount`. Normally this can be circumvented by unmounting then remounting, but I also use NixOS and the aforementioned command is automatically invoked whenever I switch system generation.

#### How do you think rclone should be changed to solve that?
Add the `--remount` flag to remount a file system.

#### How to use GitHub
* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

### PR DESCRIPTION
**Pull request by KDreynolds**
#### What is the purpose of this change?

Added a --remount option as requested in issue #6488 . option only works on linux systems. Updated documentation to reflect this optional flag.

#### Was the change discussed in an issue or in the forum before?

Issue #6488 

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by KDreynolds** [comment]
Took a crack at addressing issue #6488

File: cmd/mountlib/mount.go [hunk 1]
```diff
@@ -161,6 +163,7 @@ func AddFlags(flagSet *pflag.FlagSet) {
 	flags.BoolVarP(flagSet, &Opt.NetworkMode, "network-mode", "", Opt.NetworkMode, "Mount as remote network drive, instead of fixed disk drive (supported on Windows only)", "Mount")
 	// Unix only
 	flags.DurationVarP(flagSet, &Opt.DaemonWait, "daemon-wait", "", Opt.DaemonWait, "Time to wait for ready mount from daemon (maximum time on Linux, constant sleep time on OSX/BSD) (not supported on Windows)", "Mount")
+	flagSet.BoolVarP(&Opt.Remount, "remount", "", false, "Remount a file system")
```
Comment (ncw) [inline cmd/mountlib/mount.go:166]: I suggest you wrap this in a `if runtime.GOOS == "linux" { }` then the flag will only appear for linux systems.

File: cmd/mountlib/mount.go [hunk 2]
```diff
@@ -283,11 +286,21 @@ func NewMountCommand(commandName string, hidden bool, mount MountFn) *cobra.Comm
 
 // Mount the remote at mountpoint
 func (m *MountPoint) Mount() (mountDaemon *os.Process, err error) {
-
 	// Ensure sensible defaults
 	m.SetVolumeName(m.MountOpt.VolumeName)
 	m.SetDeviceName(m.MountOpt.DeviceName)
 
+	// Handle the --remount flag by attempting to unmount the mountpoint first
+	if m.MountOpt.Remount {
```
Comment (ncw) [inline cmd/mountlib/mount.go:294]: I'd probably put `if runtime.GOOS == linux && m.MountOpt.Remount {` here.

The compiler is smart enough to evaluate `runtime.GOOS == linux` at compile time then figure out it doesn't need to compile any of the stuff in the `if` statement so it saves bloating the binaries where this doesn't work.

**Review by ncw — COMMENTED** [review]
Apologies for the delay reviewing this.

This is looking great - thank you.

I put some `runtime.GOOS` suggestions inline.

**Comment by KDreynolds** [comment]
ok made the updates, not sure about this Race test that is failing on the go1.20 build, let me know if I can do anything to fix this.

**Pull request closed without merging**